### PR TITLE
Progressbar in PSF photometry

### PR DESCRIPTION
**Pull request by keflavich**
This PR adds an option to use a progressbar when doing PSF photometry.

### CODE REVIEW

File: photutils/psf/photometry.py [hunk 1]
```diff
@@ -243,14 +243,14 @@ def _make_mask(image, mask):
                               'automatically ignored.', AstropyUserWarning)
         return mask
 
-    def __call__(self, image, *, mask=None, init_guesses=None):
+    def __call__(self, image, *, mask=None, init_guesses=None, **kwargs):
```
Comment (larrybradley) [inline photutils/psf/photometry.py:246]: Why not just add the progress bar keyword?

File: photutils/psf/photometry.py [hunk 2]
```diff
@@ -401,7 +404,7 @@ def do_photometry(self, image, *, mask=None, init_guesses=None):
 
         return star_groups
 
-    def nstar(self, image, star_groups, *, mask=None):
+    def nstar(self, image, star_groups, *, mask=None, progressbar=None):
```
Comment (larrybradley) [inline photutils/psf/photometry.py:407]: For consistency with with other `photutils` functions, this keyword should be called `progress_bar`.

File: photutils/psf/photometry.py [hunk 3]
```diff
@@ -428,6 +431,11 @@ def nstar(self, image, star_groups, *, mask=None):
             a `True` value indicates the corresponding element of
             ``image`` is masked.
 
+        progressbar : progressbar
+            A progressbar (optional) to show progress over the star groups.
+            Could be an `astropy.utils.console.ProgressBar` or a `tqdm`
+            progressbar.
+
```
Comment (larrybradley) [inline photutils/psf/photometry.py:438]: Until just now, I was unaware that astropy even provided a `ProgressBar`.  I think the python community has converged on `tqdm` as the progress bar standard.  I would prefer that this simply be a True/False boolean, and if True would use a `tqdm` progress bar (`tqdm` is an optional dependency).  This makes the API consistent with the `progress_bar` options already available in `photutils` (e.g. see `deblend_sources`, `SourceFinder`, `ESPFBuilder`, `ImageDepth`).  Specifically, I've used the `tqdm.auto` progress bar, which switches from a text-based progress bar for Jupyter notebooks (see, e.g. https://github.com/astropy/photutils/blob/main/photutils/utils/depths.py#L247).

**Review by larrybradley — CHANGES_REQUESTED** [review]
Thanks, @keflavich.  Please see the inline comments.

**Comment by larrybradley** [comment]
This will also need a change log entry.

File: photutils/psf/photometry.py (hunk 3 above)
Comment (keflavich) [reply photutils/psf/photometry.py:437]: I didn't know about `tqdm.auto`; I agree that alleviates any need for user-specified progressbars in general.

**Review by larrybradley — APPROVED** [review]
Thanks, @keflavich

**Pull request merged**